compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ServiceDesugar.java

```diff
@@ -204,17 +206,21 @@ private void addMethodInvocation(Location pos, BLangSimpleVarRef varRef, BInvoka
     }
 
     void engageCustomServiceDesugar(BLangService service, SymbolEnv env) {
+        List<BType> expressionTypes = service.attachedExprs.stream().map(expression -> expression.type)
+                .collect(Collectors.toList());
         service.serviceClass.functions.stream().filter(fun -> Symbols.isFlagOn(fun.symbol.flags, Flags.RESOURCE))
-                .forEach(func -> engageCustomResourceDesugar(func, env));
+                .forEach(func -> engageCustomResourceDesugar(func, env, expressionTypes));
     }
 
-    private void engageCustomResourceDesugar(BLangFunction functionNode, SymbolEnv env) {
+    private void engageCustomResourceDesugar(BLangFunction functionNode, SymbolEnv env, List<BType> expressionTypes) {
         if (Symbols.isFlagOn(functionNode.symbol.flags, Flags.TRANSACTIONAL)) {
             BLangExpressionStmt stmt = new BLangExpressionStmt(transactionDesugar
                     .createBeginParticipantInvocation(functionNode.pos));
             ((BLangBlockFunctionBody) functionNode.body).stmts.add(0, stmt);
         }
-        httpFiltersDesugar.addHttpFilterStatementsToResource(functionNode, env);
-//        httpFiltersDesugar.addCustomAnnotationToResource(functionNode, env);
+        if (httpFiltersDesugar.isHttpPackage(expressionTypes)) {
+            httpFiltersDesugar.addFilterStatements(functionNode, env);
+//            httpFiltersDesugar.addOrderParamConfig(functionNode, env);
```

Comment:
I don't think we need these things anymore with the service changes, if so shall we completely remote this.
@chamil321 please confirm whether we need this.

Reply:
Yes, we don't need them. I'll send another PR removing this function and its related functions.

Reply:
Ok will merge this now.